src/Microsoft.Azure.CosmosRepository/Services/DefaultCosmosContainerService.cs

```diff
@@ -46,8 +46,11 @@ private async Task<Container> GetContainerAsync(Type itemType, bool forceContain
                 ItemConfiguration itemConfiguration = _cosmosItemConfigurationProvider.GetItemConfiguration(itemType);
 
                 Database database =
-                    await _cosmosClientProvider.UseClientAsync(
-                        client => client.CreateDatabaseIfNotExistsAsync(_options.DatabaseId)).ConfigureAwait(false);
+                    _options.IsAutoResourceCreationIfNotExistsEnabled
+                        ? await _cosmosClientProvider.UseClientAsync(
+                            client => client.CreateDatabaseIfNotExistsAsync(_options.DatabaseId)).ConfigureAwait(false)
+                        : await _cosmosClientProvider.UseClientAsync(
+                            client => Task.FromResult(client.GetDatabase(_options.DatabaseId))).ConfigureAwait(false);
```

Comment:
Can't we get a little closer to where the ternary is needed?

```suggestion
                    await _cosmosClientProvider.UseClientAsync(client =>
                        _options.IsAutoResourceCreationIfNotExistsEnabled
                            ? client.CreateDatabaseIfNotExistsAsync(_options.DatabaseId).ConfigureAwait(false)
                            : Task.FromResult(client.GetDatabase(_options.DatabaseId));
```